test/builder/auto_mocks_test.dart

```diff
@@ -3553,6 +3553,64 @@ void main() {
         expect(mocksContent, contains('class MockBaz extends _i1.Mock'));
         expect(mocksContent, contains('implements _i2.Baz'));
       });
+
+      test('when its a type parameter of function which returns another type',
```

Comment:
I'm not sure what you mean here by "a type parameter of function which returns another type." In this test case, the typedef appears to only be used as a _type argument_ in a class definition `class Foo extends BaseFoo<CreateBar>`.

Reply:
How about?
`typedef mocks are generated properly when it\'s a function which returns any type`

In this instance, `Bar` was being ignored becuase the return type of `CreateBar` want being resolved

Reply:
That SG, thanks!